### PR TITLE
Various fixes and improvements from LSP evaluation

### LINKED ISSUE #708: Closing windows (panels) not working correctly
Closing panels isn't working correctly while having panels in multiple windows.

# Problems
I encountered 2 problems:
1. Close-Buttons of non-main-windows don't work
2. Close-All-Button of main-window destroys non-main-windows

## Setup (Example)
 1. Open multiple panels (Code-Window, Tree-Window, etc.)
 3. Divide the tabs into several windows
panel_setup

## Close-Buttons of non-main-windows don't work
When dividing the panels into multiple windows, only the Close-Buttons of the main window are working.
secondary_panel_buttons
* red = not working
* green = working

## Close-All-Button of main-windows destroys
After closing the main-window through the Close-All-Button, secondary windows turn gray.
secondary_panels_destroyed

The gray "effect" goes away after dividing a destroyed panel into its own window.
secondary_panels_divided

### PR DESCRIPTION
**Pull request by falko17**
These are various collected changes[^1] from when I was preparing for the still ongoing [LSP evaluation](https://falko17.uber.space/evaluation).
As there is no real unifying theme behind the changes here, I'll just list each notable change below, together with the commit ID at the end (as the commit message sometimes contains additional information about the change.)

## New features
- 07ef7f6c1ee5130ed7d5863022687707e553ec1b: Missing LSP navigation options have been added to the context menu (e.g., "Show supertype").
- ea982dce283e1b142fbf1c4b6073b835884d498b: A new field "Hidden Edges" has been added to the city edge settings. Adding edge types in here will lead to edges of that type being hidden on startup, and only showing up upon hovering over them.
- 0619cb766dc049b1fef1301a2988d12b72b23280: The previously ignored "are selectable" setting for edges now actually leads to edges not being selectable anymore. Note that hidden edges (see above) are automatically not selectable, as the invisible edges would otherwise interfere with hovering behavior.
- abe01125f7e7dedc8ecec323164be82f29fc6ae2: Transitive edge animations have been added. When enabled (configurable in edge animation settings for source/target transitivity), hovering over a node iteratively animates in not only the directly connected edges of the node, but also recursively all sources/targets of those nodes.

## Fixes
- 75f2ae9b7e9d17384dfc7d0ff6fb5f6b615cb5cd: The context menu will now only be displayed if the user did not drag the mouse around after right-clicking.
- 24ffb5d93d01c9ca95417607293715438e1379e9: Package hierarchies in Java projects are now properly mapped onto LSP-generated code cities.
- 3ea77eff5af46e69814d8d66358c467f09b0b943: Previously, the base path for graphs wasn't always set correctly for generated builds of SEE (specifically, when creating subgraphs). This has been fixed.
- 57241d0d965f4e45969dec42dfea1c0ea108d1a1: When highlighting nodes of a code city, the width/height of the spear had been swapped, leading to a flat circle instead of a proper spear. This has been fixed.
- 732dbd2174b69cdaf1855fae1fcb71bbe1c639a3: Null values in async enumerables are now handled properly, which fixes code city generation for some language servers.
- 611ef50ac81306485fef30c3590c319cfb40baa4: Fixed node label not being displayed in certain cases.
- b8629c14c080475ced008d9b7b66bb3d424ebdfe: If the metric window is smaller than usual, metric values now won't be cut off.
- 0ca54639d6c4b39f5aa74bf9e4c7c8d64a747bae: SEE (specifically, the WindowSpace component) can now deal with multiple panels. Hence, windows[^2] can be dragged off into their own panels, can be put back into their original place, etc. This fixes #708.

[^1]: These are not *all* changes that were done for the evaluation, as some were more hacky and don't belong in master. The unaltered state of the LSP evaluation version of SEE is preserved in the branch `lsp-evaluation`.
[^2]: The terminology here might be kind of confusing. **Windows** are rectangular areas of content that we (SEE developers) have implemented. For example, the Code Window, or Tree Window, might be examples of this. **Panels** are containers for these windows. A panel can contain multiple windows by spreading them out over several tabs (and with the commit above, windows can now be spread out over multiple panels as well). Finally, the **Space** (or `WindowSpace`) manages all open panels and windows for a given user. When in multiplayer mode, there may be multiple spaces—one for each player. This is how the open windows can be synchronized over the network.

### CODE REVIEW

File: Assets/SEE/Controls/Actions/ShowEdges.cs [hunk 1]
```diff
@@ -29,6 +34,23 @@ public class ShowEdges : InteractableObjectAction
...
+        /// A token that's used to cancel the transitive edge toggling.
+        /// We need to use this instead of the edge operator's built-in conflict mechanism
+        /// because the edge operator only controls a single edge, instead of the whole transitive closure.
+        /// </summary>
+        private CancellationTokenSource edgeToggleToken;
+
+        /// <summary>
+        /// The toggle that is used to determine whether an edge is currently selected.
+        /// </summary>
+        private const string EdgeIsSelectedToggle = "IsSelected";
```
Comment (koschke) [inline Assets/SEE/Controls/Actions/ShowEdges.cs:47]: All private fields must start with a lower letter, no matter whether they are constant or not.
It's the standard Axivion rule.

File: Assets/SEE/Controls/Actions/ShowEdges.cs [hunk 2]
```diff
@@ -29,6 +34,23 @@ public class ShowEdges : InteractableObjectAction
...
+
+        /// <summary>
+        /// The delay between each depth level when showing/hiding the transitive closure of edges.
```
Comment (koschke) [inline Assets/SEE/Controls/Actions/ShowEdges.cs:50]: The unit is second, I suppose? The comment should mention that.

File: Assets/SEE/DataModel/DG/IO/LSPImporter.cs [hunk 3]
```diff
@@ -345,6 +350,71 @@ IEnumerable<string> RelevantDocumentsForPath(string path)
             {
                 return relevantExtensions.SelectMany(x => Directory.EnumerateFiles(path, $"*.{x}", SearchOption.AllDirectories));
             }
+
```
Comment (koschke) [inline Assets/SEE/DataModel/DG/IO/LSPImporter.cs:353]: Should a `return` be added here, analogously to line 403?

**Review by koschke — APPROVED** [review]
Thanks for this PR. I made only very minor remarks. I'll approve it already.

File: Assets/SEE/Controls/Actions/ShowEdges.cs (hunk 2 above)
Comment (falko17) [reply Assets/SEE/Controls/Actions/ShowEdges.cs:50]: I don't think mentioning a unit here makes much sense, given that the type of this field is `TimeSpan` (which has units built into it).

File: Assets/SEE/Controls/Actions/ShowEdges.cs (hunk 1 above)
Comment (falko17) [reply Assets/SEE/Controls/Actions/ShowEdges.cs:47]: Yes, fixed it.

File: Assets/SEE/DataModel/DG/IO/LSPImporter.cs (hunk 3 above)
Comment (falko17) [reply Assets/SEE/DataModel/DG/IO/LSPImporter.cs:353]: The `return` here is already present in line 347 (i.e., before all local methods).

File: Assets/SEE/Controls/Actions/ShowEdges.cs (hunk 2 above)
Comment (koschke) [reply Assets/SEE/Controls/Actions/ShowEdges.cs:50]: Okay, I see. Thanks for the clarification.

File: Assets/SEE/DataModel/DG/IO/LSPImporter.cs (hunk 3 above)
Comment (koschke) [reply Assets/SEE/DataModel/DG/IO/LSPImporter.cs:353]: Okay, I see. Thanks for the clarification.

**Pull request merged**